Index mapping with field not indexed results in java.lang.ClassCastException: java.lang.Boolean cannot be cast to java.lang.String

## Steps to reproduce
* Create and post a mapping for an Elasticsearch index. The mapping needs to include a field that is not indexed.
```
    "field_name" : {
        "index" : false
    }
```
* In geoserver, create a store for the index. Use the REST interface (port 9200). After the store is created, try to publish the layer.

Geoserver fails with the following error
```
Caused by: java.lang.ClassCastException: java.lang.Boolean cannot be cast to java.lang.String
	at mil.nga.giat.data.elasticsearch.ElasticCompat5.isAnalyzed(ElasticCompat5.java:88)
	at mil.nga.giat.data.elasticsearch.ElasticDataStore.add(ElasticDataStore.java:340)
	at mil.nga.giat.data.elasticsearch.ElasticDataStore.walk(ElasticDataStore.java:313)
	at mil.nga.giat.data.elasticsearch.ElasticDataStore.walk(ElasticDataStore.java:310)
	at mil.nga.giat.data.elasticsearch.ElasticDataStore.getElasticAttributes(ElasticDataStore.java:150)
	at mil.nga.giat.data.elasticsearch.ElasticFeatureSource.<init>(ElasticFeatureSource.java:48)
	at mil.nga.giat.data.elasticsearch.ElasticDataStore.createFeatureSource(ElasticDataStore.java:111)
	at org.geotools.data.store.ContentDataStore.getFeatureSource(ContentDataStore.java:395)
	at mil.nga.giat.data.elasticsearch.ElasticDataStore.getFeatureSource(ElasticDataStore.java:122)
	at org.geotools.data.store.ContentDataStore.getFeatureSource(ContentDataStore.java:686)
	at org.geotools.data.store.ContentDataStore.getFeatureSource(ContentDataStore.java:103)
	at org.geoserver.catalog.CatalogBuilder.buildFeatureType(CatalogBuilder.java:333)
	at org.geoserver.web.data.layer.NewLayerPage.buildLayerInfo(NewLayerPage.java:359)
	... 121 more
```

I looks like Elasticsearch 5.x changed the value of `index` from a String (that used to be `analyzed`, `not_analyzed`, and `no`) to just a Boolean.

--- a/gt-elasticsearch-parent/gt-elasticsearch-core/src/main/java/mil/nga/giat/data/elasticsearch/ElasticCompat5.java
+++ b/gt-elasticsearch-parent/gt-elasticsearch-core/src/main/java/mil/nga/giat/data/elasticsearch/ElasticCompat5.java
@@ -85,14 +85,12 @@ public class ElasticCompat5 implements ElasticCompat {
 
     @Override
     public boolean isAnalyzed(Map<String, Object> map) {
+        boolean analyzed = false;
         Object value = map.get("type");
-        if (value != null && Map.class.isAssignableFrom(value.getClass())) {
-            //map is not property map
-            return false;
-        } else {
-            final String type = (String) map.get("type");
-            return type != null && type.equals("text");
+        if (value != null && value instanceof String && ((String) value).equals("text")) {
+            analyzed = true;
         }
+        return analyzed;
     }
 
     @Override

--- a/gt-elasticsearch-parent/gt-elasticsearch-core/src/main/java/mil/nga/giat/data/elasticsearch/ElasticCompat5.java
+++ b/gt-elasticsearch-parent/gt-elasticsearch-core/src/main/java/mil/nga/giat/data/elasticsearch/ElasticCompat5.java
@@ -85,8 +85,14 @@ public class ElasticCompat5 implements ElasticCompat {
 
     @Override
     public boolean isAnalyzed(Map<String, Object> map) {
-        final String index = (String) map.get("index");
-        return index != null && index.equals("analyzed");
+        Object value = map.get("type");
+        if (value != null && Map.class.isAssignableFrom(value.getClass())) {
+            //map is not property map
+            return false;
+        } else {
+            final String type = (String) map.get("type");
+            return type != null && type.equals("text");
+        }
     }
 
     @Override

--- a/gt-elasticsearch-parent/gt-elasticsearch-core/src/test/java/mil/nga/giat/data/elasticsearch/ElasticCompatTest.java
+++ b/gt-elasticsearch-parent/gt-elasticsearch-core/src/test/java/mil/nga/giat/data/elasticsearch/ElasticCompatTest.java
@@ -49,10 +49,12 @@ public class ElasticCompatTest {
         assertEquals(1, point.getLon(), 1e-4);
         assertEquals(86461000, compat.parseDateTime("1970-01-02 00:01:01", "yyyy-mm-dd HH:mm:ss").getTime());
         assertFalse(compat.isAnalyzed(new HashMap<>()));
-        assertFalse(compat.isAnalyzed(ImmutableMap.of("index", "not_analyzed")));
-        assertFalse(compat.isAnalyzed(ImmutableMap.of("index", "not_valid")));
-        assertTrue(compat.isAnalyzed(ImmutableMap.of("index", "analyzed")));
+        assertFalse(compat.isAnalyzed(ImmutableMap.of("type", "keyword")));
+        assertFalse(compat.isAnalyzed(ImmutableMap.of("type", ImmutableMap.of("type", "keyword"))));
+        assertFalse(compat.isAnalyzed(ImmutableMap.of("type", "not_valid")));
+        assertTrue(compat.isAnalyzed(ImmutableMap.of("type", "text")));
     }
+    
 
     @Test(expected=IOException.class)
     public void testCreateClient() throws IOException {

